File: golang/2488. 统计中位数为 K 的子数组.go
func countSubarrays(nums []int, k int) (res int) {
    n := len(nums)
    ind := -1
    count := make(map[int]int, n+1)
    count[0] = 1
    for i, num := range nums {
        if num == k {
            ind = i
        }
    } 
    sum := 0
    for i, num := range nums {
        if num > k {
            num = 1
        } else if num < k {
            num = -1
        } else if num == k {
            num = 0
        }
        sum += num
        if i < ind {
            count[sum]++
        } else {
            res += count[sum] + count[sum-1]
        }
    }
    return
}

File: 刷题/cpp/2488. 统计中位数为 K 的子数组.cpp
class Solution {
public:
    int countSubarrays(vector<int>& nums, int k) {
        int n=nums.size();
        int ind=-1;
        int res=0;
        unordered_map<int,int> count;
        count[0]=1;
        for(int i=0;i<n;i++){
            if(nums[i]>k){
                nums[i]=1;
            }else if(nums[i]<k){
                nums[i]=-1;
            }else if(nums[i]==k){
                nums[i]=0;
                ind=i;
            }
        }
        int sum=0;
        for(int i=0;i<n;i++){
            sum+=nums[i];
            if (i<ind){
                count[sum]++;
            }else{
                res+=count[sum]+count[sum-1];
            }
        }
        return res;
    }
};